react/components/RenderProvider.tsx

```diff
@@ -654,6 +654,19 @@ export class RenderProvider extends Component<
     }
   }
 
+  private mergeRouteParams<T extends { params: any }>(
+    matchingPage: T,
+    transientRoute: { params: any }
+  ): T {
```

Comment:
We can be more accurate in typing the return saying that params are an object?

Reply:
I thought about it but this is how params it's typed on the rest of the code, so 🤷